src/rabbit_db_lvc_exchange.erl

```diff
@@ -0,0 +1,120 @@
+-module(rabbit_db_lvc_exchange).
+
+-include_lib("rabbit_common/include/rabbit.hrl").
+-include_lib("khepri/include/khepri.hrl").
+-include_lib("rabbit/include/mc.hrl").
+-include("rabbit_lvc_plugin.hrl").
+
+-export([setup_schema/0,
+         get/2,
+         insert/3,
+         delete/0,
+         delete/2
+        ]).
+
+-export([khepri_lvc_path/1, khepri_lvc_path/2]).
+
+-rabbit_mnesia_tables_to_khepri_db(
+   [{?LVC_TABLE, rabbit_lvc_plugin_m2k_converter}]).
+
+setup_schema() ->
+    rabbit_khepri:handle_fallback(
+        #{mnesia => fun() -> setup_schema_in_mnesia() end,
+          khepri => fun() -> ok end}
+    ).
+
+setup_schema_in_mnesia() ->
+    _ = mnesia:create_table(?LVC_TABLE,
+                            [{attributes, record_info(fields, cached)},
+                             {record_name, cached},
+                             {type, set},
+                             {disc_copies, [node()]}]),
+    _ = mnesia:add_table_copy(?LVC_TABLE, node(), disc_copies),
+    rabbit_table:wait([?LVC_TABLE]),
+    ok.
+
+get(XName, RoutingKey) ->
+    rabbit_khepri:handle_fallback(
+      #{mnesia => fun() -> get_in_mnesia(XName, RoutingKey) end,
+        khepri => fun() -> get_in_khepri(XName, RoutingKey) end}
+    ).
+
+get_in_mnesia(XName, RoutingKey) ->
+    case mnesia:dirty_read(
+           ?LVC_TABLE,
+           #cachekey{exchange = XName,
+                     routing_key = RoutingKey }) of
+        [] ->
+            not_found;
+        [#cached{content = Msg}] ->
+            Msg
+    end.
+
+get_in_khepri(XName, RoutingKey) ->
+    case rabbit_khepri:get(khepri_lvc_path(XName, RoutingKey)) of
+        {ok, Msg} ->
+            Msg;
+        _ ->
+            not_found
+    end.
+
+insert(XName, RKs, Msg) ->
+    rabbit_khepri:handle_fallback(
+      #{mnesia => fun() -> insert_in_mnesia(XName, RKs, Msg) end,
+        khepri => fun() -> insert_in_khepri(XName, RKs, Msg) end}
+    ).
+
+insert_in_mnesia(XName, RKs, Msg) ->
+    rabbit_mnesia:execute_mnesia_transaction(
+      fun () ->
+              [mnesia:write(?LVC_TABLE,
+                            #cached{key = #cachekey{exchange=XName,
+                                                    routing_key=K},
+                                    content = Msg},
+                            write) ||
+               K <- RKs]
+      end),
+    ok.
+
+insert_in_khepri(XName, RKs, Msg) ->
+    [ rabbit_khepri:put(khepri_lvc_path(XName, RK), Msg)
+      || RK <- RKs ],
+    ok.
+
+delete() ->
+    rabbit_khepri:handle_fallback(
+        #{mnesia => fun() -> _ = mnesia:delete_table(?LVC_TABLE), ok end,
+          khepri => fun() -> _ = rabbit_khepri:delete([rabbit_exchange_type_lvc]), ok end}
```

Comment:
`khepri_lvc_path/1/2` puts data under `?MODULE`, to have things under one place could create a `khepri_lvc_path/0` variant. The root probably shouldn't be `rabbit_db_lvc_exchange` but `rabbit_exchange_type_lvc` or something similar.

(even better would be from this point to implement them dependently ie
```
khepri_lvc_path(#resource{virtual_host = VHost, name = Name}) ->
    khepri_lvc_path() ++ [VHost, Name].
khepri_lvc_path(Exchange, RK) ->
    khepri_lvc_path(Exchange) ++ [RK].
```

another alternative: the recent history exchange uses a common exchange subtree:
```
khepri_recent_history_path(VHost, Name) ->
    ExchangePath = rabbit_db_exchange:khepri_exchange_path(VHost, Name),
    ExchangePath ++ [recent_history].
```

Reply:
to have things under one place, use `khepri_lvc_path/0` on line 86 too!